llms/hf_llm/models.py

```diff
@@ -73,8 +82,17 @@ def __init__(self, args: ModelArgs):
         self.k_proj = nn.Linear(dim, n_kv_heads * head_dim, bias=False)
         self.v_proj = nn.Linear(dim, n_kv_heads * head_dim, bias=False)
         self.o_proj = nn.Linear(n_heads * head_dim, dim, bias=False)
+        rope_scale = (
+            1 / args.rope_scaling.get("factor", 1.0)
+            if args.rope_scaling is not None
+            and args.rope_scaling.get("type") == "linear"
```

Comment:
Are there other scaling types for RoPE?

Reply:
From the Transformers source code, it seems like there is another type called `dynamic`, but I haven't seen it in any of the model's configurations.
https://github.com/huggingface/transformers/blob/main/src/transformers/models/llama/modeling_llama.py#L338

Reply:
Oh great, thanks!